packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -4237,16 +4254,18 @@ export class AdapterClass extends EventEmitter {
         }
 
         adapterObjects.getObject(options.id, options, (err, obj) => {
-            const adapterName = this.namespace.split('.')[0];
-            // remove protectedNative if not admin or own adapter
+            // remove protectedNative if not admin, not cloud or not own adapter
             if (
                 obj &&
                 'protectedNative' in obj &&
                 Array.isArray(obj.protectedNative) &&
                 obj._id &&
                 obj._id.startsWith('system.adapter.') &&
-                adapterName !== 'admin' &&
-                adapterName !== obj._id.split('.')[2]
+                obj.native &&
+                this.name !== 'admin' &&
+                this.name !== 'cloud' &&
```

Comment:
maybe better a global Array of the adapters then we can make `!NO_PROTECT_ADAPTERS.includes(this.name)` or maybe a helper `needsProtection(requestingAdapter: string)`

Reply:
would support this idea ... would be more central

Reply:
@foxriver76 Changed